core/src/main/java/de/jplag/highlight_extraction/CompleteMatchesStrategy.java

```diff
@@ -0,0 +1,39 @@
+package de.jplag.highlight_extraction;
+
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Strategy that calculates the frequencies of matches over all submissions.
+ */
+
+public class CompleteMatchesStrategy implements FrequencyStrategy {
+
+    /**
+     * Adds all matches to a map using the token sequence as the key.
+     * @param tokens List of tokens representing the match.
+     * @param comparisonId Identifier for the comparison.
+     * @param map Map that associates token subsequences with how often they occur across comparisons.
+     * @param size The minimum sub length considered in other strategies.
+     */
+
+    @Override
+    public void create(List<String> tokens, String comparisonId, Map<String, List<String>> map, int size) {
```

Comment:
We need to find more suitable representations in these data structures here. Currently, you are encoding most information with strings. For the token list `tokens`, the type should be `List<Token>` throughout your code. The map should also be mapping from and to specific types instead of strings. For the key, you want to check if the token sequence is identical. Instead of concatenating strings, use the token types. You could, for example, see if you can work with Objects.hash(). Or map the list to an immutable list of token types, then you can use that as a key (the second solution is probably better). For the values, also do not use a list of strings; instead, why not just keep an integer for the frequency?